kyo-core/shared/src/test/scala/kyoTest/abortsTest.scala

```diff
@@ -298,16 +298,16 @@ class abortsTest extends KyoTest:
                     }
                 "success" in {
                     assert(
-                        Envs[Int].run(2)(
-                            Aborts.run[Ex1](Aborts.catching[Ex1](test(Envs[Int].get)))
+                        Envs.run(2)(
+                            Aborts.run[Ex1](Aborts.catching[Ex1](test(Envs.get[Int])))
                         ).pure ==
                             Right(5)
                     )
                 }
                 "failure" in {
                     assert(
-                        Envs[Int].run(0)(
-                            Aborts.run[Ex1](Aborts.catching[Ex1](test(Envs[Int].get)))
+                        Envs.run(0)(
+                            Aborts.run[Ex1](Aborts.catching[Ex1](test(Envs.get[Int])))
```

Comment:
Given the parameter includes the type: `def test(v: Int < Env[Int]`
Can you try invoking it like: `test(Envs.get)`? not that this is particularly safe...

Reply:
yep! that works too

Reply:
Type unsafe 🙂 - inference could change depending on how the function is defined. Obviously not a problem with Kyo's encoding.

It's likely not a concern for most users, as they should rarely define functions where the result type is generic.

eg

```scala
type Pending = Envs[Float] & Envs[Double]
def myFunc[T](effect: T < Pending) = effect.map(_ / 2)

val result: ??? = myFunc(Envs.get)
```

Determinism in the Scala compiler is generally good, but I generally prefer not to take changes.

Reply:
I tested this and it's actually unsafe as you suspected. It's not much about the method inference but Scala infers a type intersection as the environment type, which fails when handling the effect. I think we'll need to add a constraint similar to `Flat` to validate that the type is not an intersection. Another option is using `Tag` to check types during handling but that's too expensive.

I added a couple pending tests because I think this doesn't seem a blocker.